### DISCUSSION TITLE
More .razor pages per module (in same client project)

### DISCUSSION THREAD

**Discussion by artmedia** [oqtane/oqtane.framework#1035] [Q&A] [answered]
Hello,
I created an external module (VS Solution).
In Client project I've Index.razor which is mapped in the Oqtane page.
Is it possible to create other .razor pages in the same VS Client project? 
I'd like to have more than one page in Oqtane which map different .razor pages for the same VS project; for example ecommerce catalog page and back-office page which share same models (so is better to have them in the same VS project) but are in different pages in Oqtane site.

**Comment by hishamco** [comment]
The `Index.razor` in the external component is one of the components in your module which will be hosted in the Oqtane Framework, there's no mapping here

Please ellaborate if you mean something else

**Reply by artmedia** [reply]
May I have Index2.razor in my VS project (external)?
It will be hosted in the Oqtane Framework as well? And how?

**Reply by hishamco** [reply]
You could have many components as much as you want

> It will be hosted in the Oqtane Framework as well? And how?

They will be hosted automatically once you places your module inside a page

**Reply by artmedia** [reply]
It seems that only Index.razor is hosted automatically.
Any other component is not hosted automatically unless they are child of Index.razor.
If you try to create Index2.razor (not child component) in the same VS project where you have Index.razor, how can you view it in Oqtane pages?
I tryed to create a new page, in Module Management select "add New Module", "Common Modules", selected my module (which is already hosted in another page) then "Add Module To Page" 
but it automatically adds Index.razor.
For Index2.razor (not child component) there is not option to be added in a page.
![Immagine](https://user-images.githubusercontent.com/4210561/103483210-fa78c380-4de5-11eb-835a-5a6ca117d3e1.png)

**Reply by hishamco** [reply]
> For Index2.razor (not child component) there is not option to be added in a page.

AFAIK know the `Index.razor` will be hosted, all other pages should be children of that page, @sbwalker correct me if I'm wrong

**Comment by sbwalker** [comment, marked as answer]
Within your module client project you simply need to organize your various components by folder ( exactly the same way that the Oqtane.Client project is organized ):

/YourModule.Client
../ModuleName1
....Index.razor
....Edit.razor
../ModuleName2
....Index.razor

Use the namespace directive in your components if you want to be explicit about naming ( otherwise Blazor will generate a default namespace ):

@namespace YourModule.ModuleName1
@inherits ModuleBase

@namespace YourModule.ModuleName2
@inherits ModuleBase

Each folder containing an Index.razor will be treated as a module that can be added to a page using the Control Panel. All components in your client project can share the same APIs and Models from the other projects in your solution.

**Reply by artmedia** [reply]
thanks
